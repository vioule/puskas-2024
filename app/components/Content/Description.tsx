import { DataType } from "../DataProvider";

export default function Description({ data }: { data: DataType }) {
  return (
    <div id="description" className="hidden">
      <div className="absolute top-[-2.7rem] h-[2.7rem] left-0 w-full uppercase font-roboto overflow-hidden">
        <div className="absolute top-0 left-0 w-full bg-black h-[2.7rem] font-black text-white tracking-[0.4rem]">
          <span
            className="block absolute top-[110%] left-0 w-full height-full pl-1"
            id="name"
          >
            {data.name}
          </span>
        </div>
        <div className="absolute bottom-0 left-0 w-full bg-black h-[1.2rem] text-gold text-[0.6rem] tracking-[0.2rem]">
          <span
            className="block absolute top-[110%] left-0 w-full height-full pl-1"
            id="comp"
          >
            {data.comp}
          </span>
        </div>
      </div>
      <div className="absolute bottom-[-1.5rem] h-[1.5rem] right-0 w-full font-rocksalt overflow-hidden">
        <div className="absolute top-0 right-0 w-full bg-black h-[2.7rem] font-normal text-white text-[0.6rem] tracking-[0.4rem]">
          <span
            className="block absolute top-[-110%] left-0 w-full height-full pr-1 text-right pt-2"
            id="year"
          >
            {data.year}
          </span>
        </div>
      </div>
    </div>
  );
}
