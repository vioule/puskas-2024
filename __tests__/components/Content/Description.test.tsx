import { render, screen } from "@testing-library/react";
import Description from "@/app/components/Content/Description";

const data = {
  year: 2009,
  name: "cristiano ronaldo",
  comp: "UEFA champions league",
};

test("render Description unchanged", () => {
  const { container } = render(<Description data={data} />);
  expect(container).toMatchSnapshot();
});

test("should have correct data", () => {
  render(<Description data={data} />);
  expect(screen.getByText(data.year)).toBeInTheDocument();
  expect(screen.getByText(data.name)).toBeInTheDocument();
  expect(screen.getByText(data.comp)).toBeInTheDocument();
});
