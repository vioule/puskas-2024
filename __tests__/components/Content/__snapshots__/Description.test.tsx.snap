// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Description unchanged 1`] = `
<div>
  <div
    class="hidden"
    id="description"
  >
    <div
      class="absolute top-[-2.7rem] h-[2.7rem] left-0 w-full uppercase font-roboto overflow-hidden"
    >
      <div
        class="absolute top-0 left-0 w-full bg-black h-[2.7rem] font-black text-white tracking-[0.4rem]"
      >
        <span
          class="block absolute top-[110%] left-0 w-full height-full pl-1"
          id="name"
        >
          cristiano ronaldo
        </span>
      </div>
      <div
        class="absolute bottom-0 left-0 w-full bg-black h-[1.2rem] text-gold text-[0.6rem] tracking-[0.2rem]"
      >
        <span
          class="block absolute top-[110%] left-0 w-full height-full pl-1"
          id="comp"
        >
          UEFA champions league
        </span>
      </div>
    </div>
    <div
      class="absolute bottom-[-1.5rem] h-[1.5rem] right-0 w-full font-rocksalt overflow-hidden"
    >
      <div
        class="absolute top-0 right-0 w-full bg-black h-[2.7rem] font-normal text-white text-[0.6rem] tracking-[0.4rem]"
      >
        <span
          class="block absolute top-[-110%] left-0 w-full height-full pr-1 text-right pt-2"
          id="year"
        >
          2009
        </span>
      </div>
    </div>
  </div>
</div>
`;
